scripts/prepare_sector_network.py

```diff
@@ -1479,89 +1514,147 @@ def add_land_transport(n, costs):
-            p_max_pu=avail_profile[nodes],
-            efficiency=options.get("bev_charge_efficiency", 0.9),
-        )
+        # Only when v2g option is True, add the v2g link
+        if options["v2g"]:
+            n.madd(
+                "Link",
+                nodes,
+                suffix=" V2G",
+                bus1=nodes,
+                bus0=nodes + " EV battery",
+                p_nom_extendable = True,
+                carrier="V2G",
+                p_max_pu=avail_profile[nodes],
+                efficiency=options.get("bev_charge_efficiency", 0.9),
+                lifetime = costs.at['Battery electric (passenger cars)', 'lifetime'], 
+            )
 
-    if electric_share > 0 and options["bev_dsm"]:
-        e_nom = (
-            number_cars
-            * options.get("bev_energy", 0.05)
-            * options["bev_availability"]
-            * electric_share
-        )
 
+        # add bev management when enabled
+        n.add("Carrier", "EV battery storage")
+
+        if options["bev_dsm"]:
+            n.madd(
+                "Store",
+                nodes,
+                suffix=" EV battery storage",
+                bus=nodes + " EV battery",
+                carrier="EV battery storage",
+                e_cyclic=True,
+                e_nom_extendable=True,
+                e_max_pu=1,
+                e_min_pu=dsm_profile[nodes],
+                lifetime = costs.at['Battery electric (passenger cars)', 'lifetime'], 
+                #lifetime?
+            )
+        
         n.madd(
-            "Store",
+            "Link",
             nodes,
-            suffix=" battery storage",
-            bus=nodes + " EV battery",
-            carrier="battery storage",
-            e_cyclic=True,
-            e_nom=e_nom,
-            e_max_pu=1,
-            e_min_pu=dsm_profile[nodes],
+            suffix=" land transport EV",
+            bus0=nodes + " EV battery",                              
+            bus1 =nodes + " land transport",
+            carrier="land transport EV",
+            lifetime = costs.at['Battery electric (passenger cars)', 'lifetime'], 
+            capital_cost = costs.at["Battery electric (passenger cars)", "fixed"]/options['EV_consumption_1car'],
+            efficiency = 1, #costs.at['Battery electric (passenger cars)', 'efficiency'],     #efficiency already accounted for in build_transport_demand  
+            p_min_pu = p_set/p_set.max(axis=0),
+            p_max_pu = p_set/p_set.max(axis=0),
+            p_nom_extendable=True,
         )
-
-    if fuel_cell_share > 0:
+        
+        #exogenous pathway if electric share is defined
+        if float(electric_share or 0) > 0:
+            # set p_nom to fulfill electric share, capital cost to 0
+            n.links.loc[n.links.carrier=='land transport EV', "p_nom_extendable"] = False
+
+            p_nom = number_cars * options.get("bev_charge_rate", 0.011) * electric_share #electric_share * p_set.max(axis=0)
```

Comment:
This link represents the "capacity" of EVs to supply transport demand.
However, the current definition represents the capacity for chargin the cars 
sinc 'options.get("bev_charge_rate", 0.011)'